File: web/src/dtos/Courier.ts
import Common from "../utility/Common";
import { DeliveryPlanRes } from "./DeliveryPlan";
import { PagedReq } from "./PagedReq";

export class CourierRes {
  courierId?: number;
  name?: string;
  description?: string;

  deliveryPlans?: DeliveryPlanRes[];
}

export class CourierResWithDeliveryPlansCount extends CourierRes {
  deliveryPlansCount?: number;
}

export class CourierReqEdit {
  name?: string = "";
  description?: string = "";
}

export class CourierReqSearch extends PagedReq {
  constructor({
    pageNumber = 1,
    pageSize = Common.DEFAULT_PAGE_SIZE,
    orderBy,
    searchText = "",
  }: PagedReq) {
    super({
      pageNumber: pageNumber,
      pageSize: pageSize,
      orderBy: orderBy,
      searchText: searchText,
    });
  }
}

File: data/Dtos/Courier.cs
﻿using data.Entities;
using data.Utility.Paging;
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace data.Dtos
{
    public class CourierRes
    {
        public int CourierId { get; set; }
        public string? Name { get; set; }
        public string? Description { get; set; }

        public IEnumerable<DeliveryPlan>? DeliveryPlans { get; set; }
    }

    public class CourierResWithDeliveryPlansCount : CourierRes
    {
        public int DeliveryPlansCount { get; set; }
    }

    public class CourierReqEdit
    {
        [Required, MaxLength(50)]
        public string? Name { get; set; }
        public string? Description { get; set; }
    }

    public class CourierReqSearch : PagedRequestDto
    {

    }
}
